File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.cpp
class Solution {
public:
    string reverseWords(string s) {
        string str = "";
        string word = "";

        for(int i = 0; i < s.size(); i++) {
            if(s[i] != ' ') word = s[i] + word;
            else {
                str += word + " ";
                word = "";
            }
        }
        str += word;


        return str;
    }
};

File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    let str = "";
    let word = "";

    for(let i = 0; i < s.length; i++) {
        if(s[i] !== ' ') word = s[i] + word;
        else {
            str += word + " ";
            word = "";
        }
    }
    str += word;
    

    return str;
};